javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -64,6 +80,133 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+                                        writeModelAssertion(methodBlock, nodeVisitor, (ClassType) responseType.getTypeArguments()[0], (Map<String, Object>) firstItem, "firstItem");
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else if (ClientModelUtil.isClientModel(returnType) && response.getBody() instanceof Map) {
+                    // Client Model
+                    // TODO(xiaofei) when return type is boolean (exists through HEAD), or Map/List and primitive types(are they possible?)
+                    writeModelAssertion(methodBlock, nodeVisitor, (ClassType) returnType, (Map<String, Object>) response.getBody(), "response");
+                }
+            } else {
+                methodBlock.line("Assertions.assertNotNull(response);");
+            }
+        };
+    }
+
+    private void addClientModelImports(IType type) {
+        ClientModel clientModel = null;
+        if (type instanceof GenericType) {
+            GenericType responseType = (GenericType) type;
+            if (PagedIterable.class.getSimpleName().equals(responseType.getName())) {
+                if (ClientModelUtil.isClientModel(responseType.getTypeArguments()[0])) {
+                    clientModel = ClientModelUtil.getClientModel(((ClassType) responseType.getTypeArguments()[0]).getName());
+                }
+            }
+        } else if (type instanceof ClassType && ClientModelUtil.isClientModel(type)) {
+            clientModel = ClientModelUtil.getClientModel(((ClassType) type).getName());
+        }
+        if (clientModel != null) {
+            clientModel.addImportsTo(this.imports, JavaSettings.getInstance());
+        }
+    }
+
+    private void writeModelAssertion(JavaBlock methodBlock, ModelExampleWriter.ExampleNodeModelInitializationVisitor nodeVisitor, ClassType modelType, Map<String, Object> body, String variableReference) {
+        methodBlock.line(String.format("Assertions.assertNotNull(%s);", variableReference));
+        ClientModel clientModel = ClientModelUtil.getClientModel(modelType.getName());
+        if (clientModel.getProperties() != null) {
+            for (ClientModelProperty property : clientModel.getProperties()) {
+                String serializedName = property.getSerializedName();
+                Object value = body.get(serializedName);
+                if (value != null) {
+                    String propertyReference = String.format("%s.%s()", variableReference, property.getGetterName());
+                    if ((property.getClientType() instanceof ClassType
+                            && !ClientModelUtil.isClientModel(property.getClientType()))
+                            || property.getClientType() instanceof PrimitiveType
+                            || property.getClientType() instanceof MapType) {
```

Comment:
Why MapType? I assume the value could be a model?

If hard to write, assertNotNull is fine.

Reply:
Ah, got it, will use assertNotNull for MapType here. 

There are also EnumType and ArrayType(for base64), I'll change the condition to exclusion instead of inclusion.

Reply:
EnumType may be good for an equal assert (if it not much effort). Bytes definitely only need notNull.